### ISSUE TITLE
[BUG] SymmetricFourierTransform in 1D unstable

### ISSUE THREAD

**Issue by MuellerSeb** [steven-murray/hankel#29]
**Describe the bug**
1D SFT is not stable at origin.

**To Reproduce**

```python
import numpy as np
import hankel

f = lambda x: 1 / (x ** 2 + 1)
k = np.linspace(0, 1, 10)
sft = hankel.SymmetricFourierTransform(ndim=1, a=-1)
sft.transform(f, k, ret_err=False)
```

This gives 2 Runtimewarnings:
```
/usr/local/lib/python3.6/dist-packages/hankel/hankel.py:158: RuntimeWarning: divide by zero encountered in true_divide
  self._f(f, np.divide.outer(self.x, k).T) * self.x ** self._x_power
/usr/local/lib/python3.6/dist-packages/hankel/hankel.py:215: RuntimeWarning: invalid value encountered in true_divide
  ret = norm * np.sum(summation, axis=-1) / knorm
```

And produces a ``nan`` value at the origin (also the second one already went bad):

```
array([       nan, 0.60363604, 0.4421674 , 0.37198499, 0.32967673,
       0.29575168, 0.26552536, 0.23805586, 0.21317494, 0.19077895])
```

**Expected behavior**
It should return 0.5 at the origin.

**Additional context**
This happens because the SFT is calculated by the hankel transformation and in case of 1D we result in a term like `` r / r`` where the integrand should only be ``f(x) * cos(k*x)``.

I would suggest calculating this case separately without using the hankel transform or adopting it.
This could also be achieved, if the alternative definition of the hankel transform would be provided: 
- https://en.wikipedia.org/wiki/Hankel_transform#Alternative_definition

There the cases for nu = -1/2 and nu = 1/2 have to be treated separately here:

https://github.com/steven-murray/hankel/blob/22f6b5632e962c3fbe425deee6492ce814541388/hankel/hankel.py#L119

Maybe I can come up with a PR about that.

**Comparison**

Something like the following already works better, but is much slower:
```python
import numpy as np
from scipy.integrate import quad
def sft_1d(f, k):
    k = np.array(k)
    res = np.empty_like(k)
    int_f = lambda r, kk: f(r) * np.cos(kk * r)
    for idx, kk in np.ndenumerate(k):
        res[idx] = quad(int_f, 0, np.inf, args=(kk,))[0] / np.pi
    return res
```

This gives:
```
>>> sft_1d(f, k)
array([0.5       , 0.44741016, 0.40043132, 0.3583185 , 0.3205619 ,
       0.28691017, 0.25676275, 0.22965815, 0.2055261 , 0.18388685])
```

**Comment by steven-murray** [comment]
Nice catch. I think this is actually a fundamental problem for all transforms done with `hankel` for `k=0`. The way that it does them is to do a substitution `x = kr` so that the transform becomes

![image](https://user-images.githubusercontent.com/1272030/70361986-a39a6680-1840-11ea-96ee-8dc88bd747eb.png)

The limits here only make sense for `k > 0`. For `k=0` we have 

![image](https://user-images.githubusercontent.com/1272030/70362133-202d4500-1841-11ea-80f7-3bf1109b6396.png)

For `nu=0` or +-1/2 the bessel function is non-zero and so the integral should be done separately using this latter equation.

I think the best way would be to add a catch in the `_get_series` method for `k=0` and do an explicit integral (but only for that k). We could use `quad` for that.

Would be awesome if you have the time to make a PR!

**Comment by MuellerSeb** [comment]
After some investigation and reading the cited paper, I came to the same conclusion. The alternative definition of the hankel transform results in a slightly different substitution, but the problem stays the same, since a division by ``k`` is needed.

I'll have another look at it.

**Comment by MuellerSeb** [comment]
See #30

**Comment by steven-murray** [comment]
Fixed by #30
